### PR TITLE
[Feature:Calendar] Add year & month dropdown

### LINKED ISSUE #10242: Dropdown Menu to Manually Set Month/Year in Calendar
To prevent having to click the left or right arrows 12 times to change the year and 1-12 times to get to a different month, there could also be a dropdown menu with each month/year.

A solution would be to simply place two more dropdown menus to the left of the preexisting one that changes the viewing format for the year and month.

I can't think of any alternative solutions that would be more effective than a dropdown menu.

### PR DESCRIPTION
**Pull request by Xiayucheng1212**
fixes #10242 
### What is the new behavior?
Two new buttons added aside the current `option` button.
The year button starts from the `current_year - 100` to `current_year + 100`
截圖 2024-03-13 下午3 04 15

### CODE REVIEW

**Review by ArnavPawar — CHANGES_REQUESTED** [review]
I think the change I requested is all that is needed to be fixed for this to be good to merge

**Comment by ArnavPawar** [comment]
> I think the change I requested is all that is needed to be fixed for this to be good to merge

I just reviewed a new PR and I think it implemented this feature correctly. That one may get merged first so if you want to brainstorm with him I think there are some features that he is planning to work on to make the UI better.

**Comment by Xiayucheng1212** [comment]
> > I think the change I requested is all that is needed to be fixed for this to be good to merge
> 
> I just reviewed a new PR and I think it implemented this feature correctly. That one may get merged first so if you want to brainstorm with him I think there are some features that he is planning to work on to make the UI better.

Got it, thank you!

**Review by arougey — COMMENTED** [review]
Hey, I also created a PR with the same issue in the branch titled "Month and Year Dropdown Menus". If you want to work from there, there is an error where the Wednesday Cell is too wide. I also moved the dropdown menu from the top to the calendar which was requested by Submitty leaders. @Xiayucheng1212

File: site/app/templates/calendar/Calendar.twig [hunk 1]
```diff
@@ -49,6 +49,42 @@
         // Load the calendar to the correct day
         loadCalendar(cookie_month, cookie_year, cookie_day, x.value);
     }
+
+    function changeMonth(x) {
```
[the remark this answers is not in the document]
Comment (04vmatsibekker) [reply site/app/templates/calendar/Calendar.twig:53]: Yea I agree with this

File: site/app/templates/calendar/Calendar.twig [hunk 2]
```diff
@@ -66,6 +102,26 @@
                 <button type="button" onclick="openOptionsModal()" class="btn btn-primary">
                     <i class="fas fa-sliders"></i> Options
                 </button>
+                <select id="calendar-year-edit" onchange="changeYear(this);">
+                    {# create options based on current year #}
+                    {% for year in range(curr_year -100, curr_year +100) %}
+                        <option value="{{ year }}" {% if curr_year == year %} selected {% endif %}>{{ year }}</option>
+                    {% endfor %}
+                </select>
+                <select id="calendar-month-edit" onchange="changeMonth(this);">
```
Comment (04vmatsibekker) [inline site/app/templates/calendar/Calendar.twig:111]: Could we somehow have a list of the months, and then use a loop. Its just quite a few lines, and would be cleaner to read if we used a loop to look through each month in a list of months

**Review by 04vmatsibekker — CHANGES_REQUESTED** [review]

**Comment by ZheyuDeng** [comment]
Close PR #10246   as PR #10249 fixed the issue #10242

**Pull request closed without merging**